### PR TITLE
1357 custom request agent

### PR DESCRIPTION
**Pull request by marcellmueller**
Implements #1357

### CODE REVIEW

File: app/backend/lib/awsCommon.ts [hunk 1]
```diff
@@ -1,31 +1,39 @@
...
+
+const nodeHandler = new NodeHttpHandler({
+  httpAgent,
+  httpsAgent,
+});
 
 const awsConfig = {
   region: AWS_S3_REGION,
   logger: console,
   httpOptions: {
```
Comment (AntBush) [inline app/backend/lib/awsCommon.ts:27]: I think the httpOptions are deprecated on v3 of the client, if anything adding the type of `S3ClientConfig` to the object can help with that. `retryDelayOptions` is also not available along with maxRetries (I think was renamed to maxAttempts)

File: app/backend/lib/awsCommon.ts [hunk 2]
```diff
@@ -1,31 +1,39 @@
...
+const httpAgent = new http.Agent({
   maxSockets: 300,
-  keepAlive: true
- });
+  keepAlive: true,
+});
+
+const nodeHandler = new NodeHttpHandler({
+  httpAgent,
+  httpsAgent,
```
Comment (AntBush) [inline app/backend/lib/awsCommon.ts:21]: Here we can set the connectionTimeout to 30 seconds rather than the awsConfig object

**Review by AntBush — CHANGES_REQUESTED** [review]

File: app/backend/lib/CustomHttpsAgent.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+class CustomHttpsAgent extends https.Agent {
+  createConnection(options, callback) {
+    let asdf;
+    const start = Date.now();
+    // @ts-ignore
+    const conn: Socket = super.createConnection(options, callback);
+    conn.on('lookup', (_err, address, _fam, host) => {
+      console.log(
+        `Currently trying to connect to: ${address} for host: ${host}`
```
[the remark this answers is not in the document]
Comment (marcellmueller) [reply app/backend/lib/CustomHttpsAgent.ts:12]: So we don't need the lookup section at all? This context would have saved me a lot of time testing this ticket.

[the remark this answers is not in the document]
Comment (AntBush) [reply app/backend/lib/CustomHttpsAgent.ts]: Oh no we do need the lookup section to save the address, but not the `console.log` here

**Review by AntBush — APPROVED** [review]

**Pull request merged**